### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by dalf**

### CODE REVIEW

**Comment by kvch** [comment]
I tried to run again the Travis build to see if it was an environment problem, but the robot tests are still failing. Could you please take a look at it?

**Comment by dalf** [comment]
This error :

```
Keyword 'Capture Page Screenshot' could not be run on failure: No browser is open
...
AssertionError: Setup failed:
WebDriverException: Message: Missing 'type' parameter
```
seems to appear in all new builds, see for example https://travis-ci.org/asciimoo/searx/builds/206807742 (A couple of fixes with search languages PR)  
 
Note : the commits are a messy (one is empty actually), sorry.

File: searx/results.py [hunk 1]
```diff
@@ -246,9 +246,12 @@ def order_results(self):
...
         for i, res in enumerate(results):
             # FIXME : handle more than one category per engine
-            category = engines[res['engine']].categories[0] + ':' + ''\
-                if 'template' not in res\
-                else res['template']
+            res['category'] = engines[res['engine']].categories[0]
+
+            # FIXME : handle more than one category per engine
+            category = engines[res['engine']].categories[0]\
+                + ':' + ('' if 'template' not in res else res['template'])\
```
Comment (asciimoo) [inline searx/results.py]: this could be `':' + res.get('template', '')`

**Review by asciimoo — CHANGES_REQUESTED** [review]
Good fixes, thanks. Please modify the concatenation according to that one minor note

**Comment by dalf** [comment]
ping ?

**Review by asciimoo — APPROVED** [review]
Sorry for the delay. Thanks.

**Pull request merged**